### DISCUSSION TITLE
Thoughts on HWID MAS Activation script for officially downloaded ISOs?

### DISCUSSION THREAD

**Discussion by ghost** [HotCakeX/Harden-Windows-Security#444] [Q&A] [answered]
Does it reduce security in any way if I activate an official Windows Pro ISO using MAS?

**Comment by HotCakeX** [comment, marked as answer]
I'm not sure, have to review every single line of code in that script and understand what they do, that's the only way to be 100% certain.

Windows licenses are cheap, especially the pro edition, and they are life time licenses, forever tied to your Microsoft account. You can use the license to reactive your OS after clean installs or complete hardware change.

**Reply by HotCakeX** [reply]
I know that repo for MAS is popular and has many stars, but like we saw in the [linux backdoor](https://techcommunity.microsoft.com/t5/microsoft-defender-vulnerability/microsoft-faq-and-guidance-for-xz-utils-backdoor/ba-p/4101961) a couple of months ago, it doesn't mean people are auditing the code all the time. Backdoors can be included in the code without people noticing.

Even if they claim some 3rd party person or entity has audited the code, what makes that 3rd party person/entity trustworthy? nothing.

So at least make sure you run it in a VM that doesn't interact with your main OS if you don't want to code review it and intend to use it.

**Reply by mketab** [reply]
You can also run the script, format your drive, and reinstall (as it ties activation to your hwid). The only issue with this is some sort of persistent malware. I reckon it would be pretty easy to find something like that if it existed.
